File: RANDOM SOLVES/CRNA1.py
import math as m
t = int(input())
while(t>0):
    n = int(input())
    x = list(map(int, input().split()))
    b=[]
    b[:n] = [0] * n
    for i in range(0,n):
        b[int(m.sqrt(x[i]))-1] += 1
    j = b.index(0)
    print((j+1)*(j+1))
    t=t-1

File: RANDOM SOLVES/CRNA1.cpp
#include <iostream> 
#include <math.h>
#include<string.h>
#include<algorithm>
using namespace std;
#define ll long long int
#define rep(i,n) for(int i=0;i<n;i++)
int main()
{
    ll t,n;
    cin>>t;
    while(t--)
    {
      cin>>n;
      ll a[n],b[n]={0};
      rep(i,n){
          cin>>a[i];
          
          b[int(round(sqrt(a[i])))-1] +=1;   
      }
      rep(i,n){
          if(b[i]==0)
          {
              cout<< (i+1)*(i+1) <<endl;
              break;
          }
      }
    }

}